troposphere/emr.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2012-2013, Antonio Alonso Dominguez <alonso.domin@gmail.com>
+# All rights reserved.
+#
+# See LICENSE file for full license.
+
+from . import AWSObject, AWSProperty, AWSHelperFn
+from .validators import (boolean, integer)
+
+
+class KeyValue(AWSHelperFn):
+    def __init__(self, key, value):
+        self.data = {'Key': key, 'Value': value}
+
+    def JSONrepr(self):
+        return self.data
+
+
+class Application(AWSProperty):
+    props = {
+        'AdditionalInfo': (dict, False),
+        'Args': (list, False),
+        'Name': (basestring, False),
+        'Version': (basestring, False)
+    }
+
+
+class BootstrapActionConfig(AWSProperty):
+    props = {
+        'Name': (basestring, True),
+        'ScriptBootstrapAction': (basestring, True)
+    }
+
+
+class ClusterConfiguration(AWSProperty):
+    props = {
+        'Classification': (basestring, False),
+        'ConfigurationProperties': (dict, False),
+        'Configurations': (list, False)
```

Comment:
This looks like it should be a list of it's own property types: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-emr-cluster-configuration.html#cfn-emr-cluster-configuration-configurations